### PR TITLE
emacs.pkgs.color-theme-solarized: convert to melpaBuild

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Description of changes

Part of  https://github.com/NixOS/nixpkgs/issues/278925

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix [hunk 1]
```diff
@@ -1,26 +1,43 @@
...
+, fetchFromGitHub
+, melpaBuild
+, writeText
 }:
 
-trivialBuild {
+let
   pname = "color-theme-solarized";
-  version = "0.pre+unstable=2017-10-24";
+  ename = "color-theme-solarized";
```
Comment (jian-lin) [inline pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix:11]: FWIW, this is the default value (`pname`) of `ename`.

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix [hunk 2]
```diff
@@ -1,26 +1,43 @@
...
-    hash = "sha256-oxX0lo6sxotEiR3nPrKPE9H01HKB3ohB/p8eEHFTp5k=";
+    rev = "b186e5d62d0b83cbf5cf38f7eb7a199dea9a3ee3";
+    hash = "sha256-7E8r56dzfD06tsQEnqU5mWSbwz9x9QPbzken2J/fhlg=";
   };
+in
+melpaBuild {
+  inherit pname ename version src;
+
+  packageRequires = [
+    emacs
```
Comment (jian-lin) [inline pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix:25]:
```suggestion
```

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -16346,12 +16346,6 @@
       fingerprint = "79A1 CC17 67C7 32B6 A8A2  BF4F 71E0 8761 642D ACD2";
     }];
   };
-  samuelrivas = {
-    email = "samuelrivas@gmail.com";
-    github = "samuelrivas";
-    githubId = 107703;
-    name = "Samuel Rivas";
-  };
```
Comment (jian-lin) [inline maintainers/maintainer-list.nix:16354]: I am not sure about this change.  I know some inactive maintainers are removed, however, I think this person is still [active](https://github.com/NixOS/nixpkgs/pulls?q=is%3Apr+author%3Asamuelrivas+).

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix [hunk 4]
```diff
@@ -1,26 +1,43 @@
...
+  '';
 
-  meta = with lib; {
+  meta =  {
     homepage = "http://ethanschoonover.com/solarized";
     description = "Precision colors for machines and people; Emacs implementation";
-    license = licenses.mit;
-    maintainers = with maintainers; [ samuelrivas AndersonTorres ];
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ AndersonTorres ];
```
Comment (jian-lin) [inline pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix:41]: Not sure about this removal too.

This person has not touched this package since [2019](https://github.com/NixOS/nixpkgs/commit/0b504bc6ff5bbb076094ee39b4656dfe058eeabd).

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (AndersonTorres) [reply maintainers/maintainer-list.nix:16354]: The most recent commit was two weeks ago. I have written this patch before that.
And before that the latest change of code was two years ago.

The best I can do now is not to remove him from maintainers list.

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix (hunk 4 above)
Comment (AndersonTorres) [reply pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix]: As I have said, they's a very sporadic contributor.
They don't even add themself as maintainer of copilot...

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (Aleksanaa) [reply maintainers/maintainer-list.nix:17414]: Have you asked this person for his opinion? If not, I'd suggest doing it in a different pr and trying to contact him.

Comment (AndersonTorres) [reply maintainers/maintainer-list.nix]: Let's do this here.

**Comment by AndersonTorres** [comment]
@samuelrivas 
Since you are a very sporadic contributor and this package requires low maintenance effort, I took the initiative of removing you from its maintainers list.

Are you still interested in being a co-maintainer of this package?

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix (hunk 4 above)
Comment (AndersonTorres) [reply pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix]: He agreed. Let's move on this.

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix [hunk 5]
```diff
@@ -1,26 +1,41 @@
...
+, writeText
 }:
 
-trivialBuild {
-  pname = "color-theme-solarized";
-  version = "0.pre+unstable=2017-10-24";
+let
+  argset = {
+    pname = "color-theme-solarized";
+    ename = "color-theme-solarized";
```
Comment (jian-lin) [inline pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix:11]: `ename` defaults to `pname`
```suggestion
```

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix [hunk 6]
```diff
@@ -1,26 +1,41 @@
...
-  src = fetchFromGitHub {
-    owner = "sellout";
-    repo = "emacs-color-theme-solarized";
-    rev = "f3ca8902ea056fb8e46cb09f09c96294e31cd4ee";
-    hash = "sha256-oxX0lo6sxotEiR3nPrKPE9H01HKB3ohB/p8eEHFTp5k=";
-  };
+    src = fetchFromGitHub {
+      owner = "sellout";
+      repo = "emacs-color-theme-solarized";
+      rev = "b186e5d62d0b83cbf5cf38f7eb7a199dea9a3ee3";
```
Comment (jian-lin) [inline pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix:17]: Ideally, the bump and the `melpaBuild` conversion should be in two separate commits.

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix [hunk 7]
```diff
@@ -1,26 +1,41 @@
...
+      hash = "sha256-7E8r56dzfD06tsQEnqU5mWSbwz9x9QPbzken2J/fhlg=";
+    };
+
+    packageRequires = [
+      color-theme
+    ];
```
Comment (jian-lin) [inline pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix:23]: Not needed since https://github.com/sellout/emacs-color-theme-solarized/commit/30dfe48d811fbbe7180bda3b98bf10e4037b6061.

```suggestion
```

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix [hunk 8]
```diff
@@ -1,26 +1,41 @@
...
+       :fetcher github)
+    '';
 
-  meta = with lib; {
-    homepage = "http://ethanschoonover.com/solarized";
-    description = "Precision colors for machines and people; Emacs implementation";
-    license = licenses.mit;
-    maintainers = with maintainers; [ samuelrivas AndersonTorres ];
+    meta =  {
+      homepage = "http://ethanschoonover.com/solarized";
```
Comment (jian-lin) [inline pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix:34]: This is not the home page for this package.  It's the home page for the Solarized theme.

And please use https.

**Review by jian-lin — COMMENTED** [review]
With https://github.com/NixOS/nixpkgs/pull/316107, this `let argset = ... in ...` refactor is not necessary.

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix (hunk 8 above)
Comment (AndersonTorres) [reply pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix:34]: THe HTTP**S** link fails sometimes. But OK, I will quote the GitHub link.

**Comment by AndersonTorres** [comment]
> With #316107, this `let argset = ... in ...` refactor is not necessary.

Let's wait this land in `master`...

**Comment by AndersonTorres** [comment]
Closing in favor of https://github.com/NixOS/nixpkgs/pull/325168
